### PR TITLE
Load Gradle dependencies lazily

### LINKED ISSUE #1822: [Feature] Lazy dependency resolution to prevent OOM on large monorepos
**Is your feature request related to a problem? Please describe.**

When opening large monorepos (700+ Gradle modules), the extension crashes with `java.lang.OutOfMemoryError: Java heap space` in `GetBuildHandler.getDependencyItem()`. On startup, the `GetBuild` gRPC call always resolves the full transitive dependency tree of all modules - `GradleProjectModelBuilder.generateDefaultGradleDependencyNode()` resolves every configuration of every subproject in the Gradle daemon, and then `getDependencyItem()` recursively serializes the entire graph into nested protobuf messages with no depth limit or pagination.

Because the gRPC task server, BSP build server, and language server all share a single JVM process, the OOM kills everything - including BSP, which users may rely on independently of the sidebar. There is no setting that prevents this: `gradle.autoDetect: "off"` does not suppress the `GetBuild` call, and `gradle.nestedProjects: false` has no effect on the dependency walk.

**Describe the solution you'd like**

Remove dependency resolution from the initial `GetBuild` call and instead load dependencies lazily per-project via a new gRPC endpoint.

Specifically:
1. Add a new RPC `GetProjectDependencies(GetProjectDependenciesRequest) returns (GetProjectDependenciesReply)` to the proto definition, accepting a `project_dir`, `project_path` (e.g. `:modules:core`), `gradle_config`, and `cancellation_key`, and returning a single `DependencyItem`.
2. Skip `generateDefaultGradleDependencyNode()` in `GradleProjectModelBuilder.buildAll()` - always return an empty root dependency node during the initial build. This will skip the expensive resolution in the Gradle daemon on startup.
3. In `GetBuildHandler.getProjectData()`, set an empty `DependencyItem` (project name only, no children) instead of calling `getDependencyItem()`.
4. Add a new `GetProjectDependenciesHandler` that connects to Gradle via the Tooling API and resolves dependencies for the single requested project path on demand.
5. On the extension side, modify `GradleDependencyProvider.getDependencies()` to call the new endpoint when the user expands a project's Dependencies tree node, instead of extracting from the cached `GradleBuild` response.

This approach means:
- Initial activation is fast regardless of project size - no dependency resolution at all.
- Dependencies are still fully available on demand (per-project scope, not all-at-once).
- OOM shouldn't be an issue any more since only a single project's dependency tree is ever in memory.
- No user-facing setting is required - it just works for all project sizes.

**Describe alternatives you've considered**

A simpler alternative is adding a `gradle.dependencies.enabled` boolean setting (default `true`) that, when set to `false`, skips dependency resolution entirely - both at the Gradle plugin level (via a system property passed to `BuildActionExecuter`) and at the gRPC serialization level (returning an empty `DependencyItem` in `getProjectData()`). This is a smaller change (~8 files, no new endpoint) and would fix it for large monorepo users, but it requires users to discover and set the option, and makes dependencies completely unavailable when disabled.

**Additional context**

The OOM stack trace:
```
java.lang.OutOfMemoryError: Java heap space
    at ...GetBuildHandler.getDependencyItem(GetBuildHandler.java:254)
```

Key code locations involved:
- `proto/gradle.proto` - `GetBuildRequest` / `DependencyItem` messages
- `gradle-server/.../handlers/GetBuildHandler.java` - `getDependencyItem()` (recursive, unbounded) and `getProjectData()` (calls it unconditionally)
- `gradle-plugin/.../GradleProjectModelBuilder.java` - `generateDefaultGradleDependencyNode()` resolves all configurations of all subprojects
- `extension/src/client/TaskServerClient.ts` - sends `GetBuildRequest` on activation
- `extension/src/dependencies/GradleDependencyProvider.ts` - extracts dependencies from cached build

The extension currently has zero safeguards (no depth limit, no pagination, no opt-out) for projects with large dependency graphs. Separating BSP into its own JVM process would also help (so a sidebar OOM can't take down BSP), but this would be a much more complex change which is not necessary imho to achieve the actual goal of not running into OOMs when opening big Monorepos.

### PR DESCRIPTION
**Pull request by chagong**
## Summary
- avoid resolving full Gradle dependency trees during initial project/task loading
- add a lazy GetProjectDependencies RPC and dependency-only Tooling API model/action
- load dependency nodes on expansion with per-node stop and reload controls
- clear dependency cache on Gradle refresh and add tests for empty/cancelled reload behavior

Fixes #1822

## Testing
- ./gradlew build
- cd extension && npm run compile:test
- cd extension && npm run lint (passes with existing no-explicit-any warnings)
- cd extension && npm run compile
- cd extension && npm test: no-gradle unit suite passed (56 passing, 1 pending); Gradle integration fixture still failed at task discovery before dependency expansion behavior was exercised

### CODE REVIEW

File: gradle-plugin-api/src/main/java/com/microsoft/gradle/api/GradleDependencyModelAction.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+				normalizedPath.append(":").append(segment);
+			}
+		}
+		return normalizedPath.length() == 0 ? ":" : normalizedPath.toString();
+	}
+
+	@Override
+	public GradleDependencyNode execute(BuildController controller) {
+		GradleBuild build = controller.getBuildModel();
+		for (BasicGradleProject project : build.getProjects()) {
```
Comment (wenytang-ms) [inline gradle-plugin-api/src/main/java/com/microsoft/gradle/api/GradleDependencyModelAction.java:38]: **Composite builds (`includeBuild`) are not handled.** `controller.getBuildModel()` only returns the root build, so `build.getProjects()` will not contain projects from any included builds. For users that wire multiple repos together via `includeBuild`, expanding the Dependencies node of any included-build subproject will fall through to `return null` at line 40 and surface as `IllegalArgumentException: Cannot find Gradle project` on the server, i.e. "Failed to load dependencies" in the tree.

Consider also iterating `build.getIncludedBuilds()` (and recursively their `getProjects()`) so the action can resolve a project path regardless of which build it lives in. This is a functional regression vs. the previous `GetBuild`-based flow that walked the model server-side.

File: gradle-server/src/main/java/com/github/badsyntax/gradle/handlers/GetProjectDependenciesHandler.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+			}
+			action.withArguments(arguments);
+			CancellationToken cancellationToken = GradleBuildCancellation.buildToken(req.getCancellationKey());
+			action.withCancellationToken(cancellationToken).setColorOutput(req.getShowOutputColors());
+			if (!Strings.isNullOrEmpty(req.getGradleConfig().getJavaHome())) {
+				action.setJavaHome(new File(req.getGradleConfig().getJavaHome()));
+			}
+			GradleDependencyNode dependencyNode = action.run();
+			if (dependencyNode == null) {
+				throw new IllegalArgumentException("Cannot find Gradle project: " + req.getProjectPath());
```
Comment (wenytang-ms) [inline gradle-server/src/main/java/com/github/badsyntax/gradle/handlers/GetProjectDependenciesHandler.java]: Throwing `IllegalArgumentException` here causes the generic `catch (Exception e)` below to send the error via `ErrorMessageBuilder.build(e)`, which surfaces as a generic gRPC `UNKNOWN`/`INTERNAL` on the client. The TS side then caches a `"Failed to load dependencies"` hint, which is indistinguishable from a real build failure.

For the "project path not found" case, prefer `responseObserver.onError(ErrorMessageBuilder.build(e, Status.NOT_FOUND))` (or just return an empty `DependencyItem`) so the extension can differentiate this from a real failure and present a better message. Bonus: combined with the composite-build fix above, this becomes the legitimate signal for "this Gradle project really doesn't exist in any build".

File: extension/src/client/TaskServerClient.ts [hunk 3]
```diff
@@ -219,6 +224,57 @@ export class TaskServerClient implements vscode.Disposable {
...
+                            resolve(undefined);
+                            return;
+                        }
+                        logger.error(
+                            `Error getting dependencies for ${normalizedProjectPath}: ${err.details || err.message}`
+                        );
+                        this.statusBarItem.command = COMMAND_SHOW_LOGS;
+                        this.statusBarItem.text = "$(warning) Gradle: Dependency Error";
+                        this.statusBarItem.show();
+                        reject(err);
```
Comment (wenytang-ms) [inline extension/src/client/TaskServerClient.ts]: `getProjectDependencies` is the only public method on this client that **rejects** on a non-cancellation error rather than `resolve(undefined)` (compare with `getBuild`, which logs + sets the status bar + resolves `undefined`). Today only `GradleDependencyProvider.loadDependencies` calls it and wraps in `try/catch`, so it works — but the inconsistency is a trap for future callers, who will silently produce unhandled promise rejections.

Two reasonable options:
1. Match `getBuild`'s pattern and `resolve(undefined)` after logging + showing the status bar warning (the loading handler already handles `undefined` as "no dependencies").
2. Keep the reject, but document on the JSDoc that callers must `.catch` and surface the contract clearly.

Option 1 is more consistent with the rest of this class.

File: gradle-server/src/main/java/com/github/badsyntax/gradle/handlers/GetProjectDependenciesHandler.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+			if ("true".equals(debugPlugin)) {
+				arguments.add("-Dorg.gradle.debug=true");
+			}
+			File initScript = PluginUtils.getInitScript();
+			if (initScript != null) {
+				arguments.addAll(Arrays.asList("--init-script", initScript.getAbsolutePath()));
+			}
+			String jvmArguments = req.getGradleConfig().getJvmArguments();
+			if (!Strings.isNullOrEmpty(jvmArguments)) {
+				arguments.addAll(Arrays.stream(jvmArguments.split(" ")).filter(e -> e != null && !e.isEmpty())
```
Comment (wenytang-ms) [inline gradle-server/src/main/java/com/github/badsyntax/gradle/handlers/GetProjectDependenciesHandler.java]: `jvmArguments.split(" ")` doesn't handle quoted arguments containing spaces — e.g. `-Dfoo="hello world"` will be torn into `-Dfoo="hello` and `world"`. This is the same pattern as `GetBuildHandler`, so it's an inherited bug rather than a new one, but since the new dependency-resolution path is the user-visible path going forward, it'd be great to extract a shared tokenizer (e.g. Apache `CommandLineUtils.translateCommandline` or an equivalent) and fix both call sites in one go.

Not a blocker for this PR if you'd rather track it separately.

**Review by wenytang-ms — COMMENTED** [review]
Thanks for tackling #1822 — the lazy loading direction is the right call. A few additional findings on top of the existing review:

File: gradle-server/src/main/java/com/github/badsyntax/gradle/handlers/DependencyItemUtils.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+		item.setName(node.getName());
+		item.setTypeValue(node.getType().ordinal());
+		if (node.getChildren() == null) {
+			return item.build();
+		}
+		List<DependencyItem> children = new ArrayList<>();
+		for (GradleDependencyNode child : node.getChildren()) {
+			children.add(getDependencyItem(child));
+		}
+		item.addAllChildren(children);
```
[the remark this answers is not in the document]
Comment (chagong) [reply gradle-server/src/main/java/com/github/badsyntax/gradle/handlers/DependencyItemUtils.java:23]: This is intentional for this PR. The OOM path was the eager initial `GetBuild`, which previously resolved dependency graphs for every project during project loading. The lazy RPC still resolves all configurations for the single expanded project, so very large individual projects can still be expensive, but the memory cost is now bounded to the user's explicit expansion instead of workspace startup.

**Review by wenytang-ms — APPROVED** [review]
